src/kibana/directives/saved_object_finder.js

```diff
@@ -104,100 +108,121 @@ define(function (require) {
           filterResults();
         });
 
-        $scope.selectedItem = false;
-        $input.on('keydown', (function () {
-          var enter = 13;
-          var up = 38;
-          var down = 40;
-          var left = 37;
-          var right = 39;
-          var esc = 27;
+        //manages the state of the keyboard selector
+        self.selector = {
+          enabled: false,
+          index: -1
+        };
 
-          var scrollIntoView = function ($el, snapTop) {
-            var el = $el[0];
+        //key handler for the filter text box
+        self.filterKeyDown = function ($event) {
+          if (keymap[$event.keyCode] !== 'tab') return;
 
-            if (!el) return;
+          if (self.hits.length === 0) return;
 
-            if ('scrollIntoViewIfNeeded' in el) {
-              el.scrollIntoViewIfNeeded(snapTop);
-            } else if ('scrollIntoView' in el) {
-              el.scrollIntoView(snapTop);
-            }
-          };
+          self.selector.index = 0;
+          self.selector.enabled = true;
 
-          return function (event) {
-            var $next;
-            var goingUp;
+          $timeout(function () {
```

Comment:
You should probably use [`$scope.evalAsync()`](https://docs.angularjs.org/api/ng/type/$rootScope.Scope#$evalAsync) here.